flink-python/pyflink/datastream/connectors/cassandra.py

```diff
@@ -0,0 +1,1648 @@
+        deployed in EC2.
+        """
+        JEC2MultiRegionAddressTranslator = get_gateway().jvm.\
+            com.datastax.driver.core.policies.EC2MultiRegionAddressTranslator
+        return AddressTranslator(JEC2MultiRegionAddressTranslator())
+
+
+class TimestampGenerator(object):
+    """
+    Generates client-side, microsecond-precision query timestamps.
+
+    Given that Cassandra uses those timestamps to resolve conflicts, implementations should generate
+    incrementing timestamps for successive implementations.
+    """
+
+    def __init__(self, j_timestamp_generator):
+        self._j_timestamp_generator = j_timestamp_generator
+
+    @staticmethod
+    def default_timestamp_generator() -> 'TimestampGenerator':
+        """
+        The default speculative retry policy.
+
+        The default speculative retry policy is a NoSpeculativeExecutionPolicy.
+        """
+        JPolicies = get_gateway().jvm.com.datastax.driver.core.policies.Policies
+        return TimestampGenerator(JPolicies.defaultTimestampGenerator())
+
+    @staticmethod
+    def atomic_monotonic_timestamp_generator() -> 'TimestampGenerator':
+        """
+        A timestamp generator that guarantees monotonically increasing timestamps among all client
+        threads, and logs warnings when timestamps drift in the future.
+        """
+        JAtomicMonotonicTimestampGenerator = get_gateway().jvm.\
+            com.datastax.driver.core.AtomicMonotonicTimestampGenerator
+        return TimestampGenerator(JAtomicMonotonicTimestampGenerator())
+
+    @staticmethod
+    def server_side_timestamp_generator() -> 'TimestampGenerator':
+        """
+        A timestamp generator that always returns Java Long.MIN_VALUE, in order to let Cassandra
+        assign server-side timestamps.
+        """
+        JServerSideTimestampGenerator = get_gateway().jvm.\
+            com.datastax.driver.core.ServerSideTimestampGenerator
+        return TimestampGenerator(JServerSideTimestampGenerator.INSTANCE)
+
+
+class CodecRegistry(object):
+    """
+    A registry for TypeCodecs.
+    """
+
+    def __init__(self, j_codec_registry):
+        self._j_codec_registry = j_codec_registry
+
+    @staticmethod
+    def default_codec_registry() -> 'CodecRegistry':
+        """
+        The default CodecRegistry instance.
+
+        It will be shared among all Cluster instances that were not explicitly built with a
+        different instance.
+        """
+        JCodecRegistry = get_gateway().jvm.com.datastax.driver.core.CodecRegistry
+        return CodecRegistry(JCodecRegistry.DEFAULT_INSTANCE)
+
+
+class AuthProvider(object):
+    """
+    Provides Authenticator instances for use when connecting to Cassandra nodes.
+    """
+
+    def __init__(self, j_auth_provider):
+        self._j_auth_provider = j_auth_provider
+
+    @staticmethod
+    def default_auth_provider(username: str, password: str) -> 'AuthProvider':
```

Comment:
```suggestion
    def plain_text_auth_provider(username: str, password: str) -> 'AuthProvider':
```

Reply:
This is the part about ClusterBuilder, I will create a new PR and fix it for easy review~